frontend/stores/orders-store.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { act } from '@testing-library/react'
+import { useOrdersStore } from './orders-store'
+
+const mockOrderData = { id: 'some-id', name: 'Test Order' }
+
+describe('Orders Store', () => {
+  const initialOrderState = useOrdersStore.getState()
+  let requestMock: jest.Mock
+
+  beforeAll(() => {
+    jest.useFakeTimers()
+    Date.now = jest.fn(() => 0)
+  })
+
+  beforeEach(() => {
+    useOrdersStore.setState(initialOrderState)
+    requestMock = jest.fn()
+  })
+
+  afterAll(() => {
+    jest.useRealTimers()
+  })
+
+  it('should set loading to true when getOrderById is called', async () => {
+    requestMock.mockImplementation(
+      () =>
+        new Promise((resolve) => {
+          Promise.resolve().then(() => {
+            resolve({ order: {} })
+          })
+        })
+    )
+
+    await act(async () => {
```

Comment:
act not needed if we await the promise correctly at the end. See `frontend/stores/assets-store.spec.ts`